### PR TITLE
Fix consistency issues with eyeball screwdrivers

### PR DESCRIPTION
**Pull request by ojacobson**
- Eyeball stabs now play an attack sound, like most other attacks.
- Nearsightedness will only be applied once, rather than every non-blinding attack (and not at all if the victim is already blind)
- Blinding will only be applied once, ditto.

Thanks @freerealestate for getting this started.

### CODE REVIEW

**Comment by freerealestate** [comment]
Jesus Christ. What an asshole.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -395,10 +397,10 @@ var/global/image/fire_overlay = image("icon" = 'icons/effects/fire.dmi', "icon_s
 	M.eye_blurry += rand(3,4)
 	M.eye_stat += rand(2,4)
 	if (M.eye_stat >= 10)
-		M.eye_blurry += 15+(0.1*M.eye_blurry)
-		M.disabilities |= NEARSIGHT
-		if(M.stat != 2)
```
Comment (Anonus) [inline code/game/objects/items.dm]: Shouldn't this still be in here?

Comment (ojacobson) [inline code/game/objects/items.dm]: This is actually the first time I've touched the code; where can I find the definition of M.stat? Looks like `mob/living/carbon`, but from there I'm lost.

You're probably right, I just want to read the rest of the code first.

Comment (ojacobson) [inline code/game/objects/items.dm]: Better?

**Comment by Chocobro** [comment]
that third commit has a funny name

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -396,9 +398,11 @@ var/global/image/fire_overlay = image("icon" = 'icons/effects/fire.dmi', "icon_s
 	M.eye_stat += rand(2,4)
 	if (M.eye_stat >= 10)
 		M.eye_blurry += 15+(0.1*M.eye_blurry)
-		M.disabilities |= NEARSIGHT
 		if(M.stat != 2)
 			M << "<span class='danger'>Your eyes start to bleed profusely!</span>"
+		if (!M.disabilities & (NEARSIGHT | BLIND))
```
Comment (phil235) [inline code/game/objects/items.dm]: I think you mean `if (!(M.disabilities & (NEARSIGHT | BLIND)))`

Comment (ojacobson) [inline code/game/objects/items.dm]: It's true.

**Comment by PKPenguin321** [comment]
:+1: hopefully people won't complain as much now

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -406,7 +410,7 @@ var/global/image/fire_overlay = image("icon" = 'icons/effects/fire.dmi', "icon_s
 			M.eye_blurry += 10
 			M.Paralyse(1)
```
Comment (freerealestate) [inline code/game/objects/items.dm]: You've removed an if check I added that made sure the person wasn't already paralysed or weakened before applying the stun. You currently have a 50% success rate of chainstunning someone.

**Comment by kingofkosmos** [comment]
Imo the M.drop_item() should go, because no other attack in the game makes you drop stuff because of "pain", making this still inconsistent.

**Comment by Fox-McCloud** [comment]
Just remove eystab altogether; it's overpowered and broken as hell.

**Comment by freerealestate** [comment]
>it's overpowered and broken as hell.

ty

**Comment by Iamgoofball** [comment]
it just needs a few nerfs IMO and making the code work with is_sharp() like proper OOP code

**Comment by freerealestate** [comment]
Didn't you say you were against nerfing this in my PR?

File: code/game/objects/items.dm [hunk 4]
```diff
@@ -396,9 +398,11 @@ var/global/image/fire_overlay = image("icon" = 'icons/effects/fire.dmi', "icon_s
 	M.eye_stat += rand(2,4)
 	if (M.eye_stat >= 10)
 		M.eye_blurry += 15+(0.1*M.eye_blurry)
-		M.disabilities |= NEARSIGHT
 		if(M.stat != 2)
 			M << "<span class='danger'>Your eyes start to bleed profusely!</span>"
```
Comment (kingofkosmos) [inline code/game/objects/items.dm]: Bleeding profusely, but no actual bleeding/bloodloss occurs.

**Comment by MrPerson** [comment]
>is_sharp()
>proper OO

HAHAHAHAHAHAHAHAHA

**Comment by Miauw62** [comment]
is_sharp() is probably the funniest proc we have

**Comment by optimumtact** [comment]
>5 commits
>6 lines changes

OP i Love you and all but please think about rebasing or amending commits in future.

**Comment by Robustin** [comment]
Ok, this looks like someone is trying to sneak in nerfs under a veil of bug-fixing.

Yes sound and animation should be added, not having them could be seen as a bug.

But it sounds like you're cutting the chain-stun potential as well. Almost every other stun in the game can be re-applied while it's still active. This isn't a bug, and removing it is just a way to make screwdrivers a less effective weapon.

**Comment by Anonus** [comment]
Please note that there are no nerfs in this pull, even if freerealestate really wants them added back in.

**Comment by Iamgoofball** [comment]
Miauw its an improvement over what it was before, anyways

**Comment by ojacobson** [comment]
@optimumtact 
> OP i Love you and all but please think about rebasing or amending commits in future.

Have a look at #10472 for some context. I didn't think it was appropriate to reuse the original author's code and erase their commits, but I'm happy to rebase the whole thing down to a single commit if you'd prefer.

**Comment by freerealestate** [comment]
Rebase it if you like.

>Bleeding profusely, but no actual bleeding/bloodloss occurs.

It's meant to tell you you've become nearsighted. That lack of clarity
behind it is why I changed it in the original PR to be more straightforward.

>But it sounds like you're cutting the chain-stun potential as well. Almost
every other stun in the game can be re-applied while it's still active.
This isn't a bug, and removing it is just a way to make screwdrivers a less
effective weapon.

They're already extremely effective. Removing the ability to keep someone
stunned indefinitely doesn't mean you don't have a 50% chance to re-stun
them while they stand back up and attempt to limp off bleeding, or go for a
disarm.

Other stun weapons have their ability to chainstun limited, anyway. You
can't do it at all with the telebaton, and stun batons have charge. So why
this commonly found weapon should have a 50% chance to chainstun after 3-5
hits is beyond me. The idea people need to put on glasses or mask up
roundstart to avoid having their eyes gouged out and spending 5 - 10
minutes at the chemistry desk trying to get enough oculine to fix their
blindness/nearsightedness/blurryness is a really retarded one.

I'm really struggling to find the source of the stupidity people are
displaying here.

**Comment by Anonus** [comment]
Why is it infinite stun? Because A: it doesn't work with even a shitty pair of glasses and B: it'll kill them if you keep doing it, thus making it ineffective if you want to hold them captive without cloning them.
Also, if you can't chainstun people with the telebaton, you're doing it wrong.

**Comment by freerealestate** [comment]
The telebaton is coded so you can't apply a stun with it while the person
is already stunned. I didn't say you can't still spam click someone with it
to stun them again when they start to get up. I was making the same change
to the screwdriver. The fact you've indicated you can still easily
chainstun someone even with that limitation shows you believe that change
wouldn't be much of a nerf at all.

The issue with the infinite stun is not "holding someone captive" (with a
screwdriver?). It's that it's a cheap as fuck way to kill someone. It takes
between 3 - 5 hits for a 50% chance each stab to chainstun someone to crit.
Even if they're brought out of it, their eyes are wrecked to the point they
need a 5 or 10 minute downtime to fix them, based on medbay's competence
and willingness to help. I'm just repeating myself here.

It's obvious it needs a nerf, and the adamant denial in the face of it is
pretty funny.

**Pull request merged**